app/main.py

```diff
@@ -1,6 +1,22 @@
+from functools import wraps
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> None:
+        func_key = func.__name__
+        key = (func_key, args, frozenset(kwargs.items()))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
+            result = func(*args, **kwargs)
+            cache_dict[key] = result
+            print("Calculating new result")
+            return result
+
+    return wrapper
```

Comment:
Remove the `py-cache-decorator.py` file from the pull request, there is no need for it.

Reply:
I am sorry, but can you explain this point because I think I don't get it quite right?

Reply:
Yes, for sure, [here is](https://github.com/mate-academy/py-cache-decorator/pull/816#discussion_r1436399474) a comment.